### DISCUSSION TITLE
Should dependency of certain jupyter notebooks be removed?

### DISCUSSION THREAD

**Discussion by maierbn** [boschresearch/pylife#42] [General]
The notebook [`demos/lifetime_calc`](https://github.com/boschresearch/pylife/blob/develop/demos/lifetime_calc.ipynb) assumes an existing file that is loaded by `%store -r rf_dict`.
This file can be created with the notebook [`time_series_handling`](https://github.com/boschresearch/pylife/blob/develop/demos/time_series_handling.ipynb), the last cell has the code `%store rf_dict`.

This seems a bit fragile to me. It is somewhat documented (lifetime_calc says `More information about the time series and loading handlin and the RF generation you can find in the notebook time_series_handling`). 

When building the docs locally, I get an error because of this (however, the automatic doc build in the Actions succeeds):

```
Notebook error:
CellExecutionError in demos/lifetime_calc.nblink:
------------------
%store -r rf_dict
------------------


---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 1
----> 1 get_ipython().run_line_magic('store', '-r rf_dict')

File ~/anaconda3/envs/pylife3.10/lib/python3.10/site-packages/IPython/core/interactiveshell.py:2454, in InteractiveShell.run_line_magic(self, magic_name, line, _stack_depth)
   2452     kwargs['local_ns'] = self.get_local_scope(stack_depth)
   2453 with self.builtin_trap:
-> 2454     result = fn(*args, **kwargs)
   2456 # The code below prevents the output from being displayed
   2457 # when using magics with decorator @output_can_be_silenced
   2458 # when the last Python token in the expression is a ';'.
   2459 if getattr(fn, magic.MAGIC_OUTPUT_CAN_BE_SILENCED, False):

File ~/anaconda3/envs/pylife3.10/lib/python3.10/site-packages/IPython/extensions/storemagic.py:148, in StoreMagics.store(self, parameter_s)
    146 for arg in args:
    147     try:
--> 148         obj = db['autorestore/' + arg]
    149     except KeyError:
    150         try:

TypeError: 'PickleShareDB' object is not subscriptable

```

Should we remove this dependency? How else could we load/generate the required data in `lifetime_calc`?

**Comment by johannes-mueller** [comment]
We used to have the pickle file in the code base. That caused also some pain. The thing is that the two notebooks from a calculation point of view do indeed depend on each other. First you calculate the load collective from time series data, then you use the load collective to do a damage calculation.

There is 9abebdd96e04f5c2660b47f1c61e9a1453bb88b4 to force that the notebooks producing stuff that other notebooks depend on during the documentation build. Maybe we can figure out why this does not work anymore.
